Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,166 @@
+// <copyright file="TreeSchemaValidatorTests.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     Tests for the TreeSchemaValidator class.
+// </summary>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Forge.TreeWalker.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Microsoft.Forge.DataContracts;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Schema;
+    using Microsoft.Forge.TreeWalker;
+
+    [TestClass]
+    public class TreeSchemaValidatorTests
+    {
+        private string forgeTreeAsString;
+        private ForgeTree treeSchema;
+        private Dictionary<string, ForgeTree> treeSchemas;
+        private string forgeTreeFromPath;
+        private string schemaDirectoryPath;
+        private string stringRules;
+        private string rulesForDictionary;
+        private JSchema jschemaRules;
+        private string invalidSchemaDirectoryPath;
+        private string invalidSchemaWithErrorContent;
+        private string forgeTreeDictionaryAsString;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            forgeTreeFromPath = Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas\\TardigradeSchema.json");
+            schemaDirectoryPath = Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas");
+            stringRules = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "contracts\\ForgeSchemaValidationRules.json"));
+            rulesForDictionary = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "contracts\\ForgeSchemaDictionaryValidationRules.json"));
+            jschemaRules = JSchema.Parse(stringRules);
+            forgeTreeAsString = File.ReadAllText(forgeTreeFromPath);
+            forgeTreeDictionaryAsString = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas\\SubroutineSchema.json"));
+            treeSchema = JsonConvert.DeserializeObject<ForgeTree>((string)forgeTreeAsString);
+            treeSchemas = new Dictionary<string, ForgeTree>();
+            treeSchemas.Add("tree1", treeSchema);
+            treeSchemas.Add("tree2", treeSchema);
+            treeSchemas.Add("tree3", treeSchema);
+            invalidSchemaWithErrorContent = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "test\\InvalidTestSchemas\\InvalidTestSchemaErrorContent.json"));
+            invalidSchemaDirectoryPath = "test\\ExampleSchemas\\TardigradeSchema.json";
+        }
+
+        [TestMethod]
+        public void Test_GetLinkedJSchemaRules()
+        {
+            try
+            {
+                JSchema linkedRules = ForgeSchemaValidator.GetLinkedJSchemaRules(stringRules, stringRules, "//ForgeSchemaValidationRules.json");
```

Comment:
>stringRules, stringRules [](http://example.com/codeflow?start=81&length=24)

You should be passing in rulesForDictionary for the parentRules.

Can you flip the ordering of the parameters. I feel like parent should go before child, not sure why. If you have a different opinion, we can discuss.